plugins/woocommerce/client/legacy/js/frontend/address-autocomplete.js

```diff
@@ -0,0 +1,1249 @@
+/**
+ * Address provider registration for WooCommerce shortcode checkout
+ */
+
+// Make functions and state available globally under window.wc.addressAutocomplete
+window.wc = window.wc || {};
+window.wc.addressAutocomplete = window.wc.addressAutocomplete || {
+	providers: {},
+	activeProvider: { billing: null, shipping: null },
+};
+
+/**
+ * Register an address autocomplete provider
+ *
+ * @param {Object} provider The provider object
+ * @return {boolean} Whether the registration was successful
+ */
+function registerAddressAutocompleteProvider( provider ) {
+	try {
+		// Check required properties
+		if ( ! provider || typeof provider !== 'object' ) {
+			throw new Error( 'Address provider must be a valid object' );
+		}
+
+		if ( ! provider.id || typeof provider.id !== 'string' ) {
+			throw new Error( 'Address provider must have a valid ID' );
+		}
+
+		if ( typeof provider.canSearch !== 'function' ) {
+			throw new Error(
+				'Address provider must have a canSearch function'
+			);
+		}
+
+		if ( typeof provider.search !== 'function' ) {
+			throw new Error( 'Address provider must have a search function' );
+		}
+
+		if ( typeof provider.select !== 'function' ) {
+			throw new Error( 'Address provider must have a select function' );
+		}
+
+		// Check if provider is registered on server.
+		var serverProviders = [];
```

Comment:
```suggestion
		let serverProviders = [];
```
I don't think you are relying on hoisting here, so `let` should be safe. I wouldn't suggest it normally, but I can see you're using `const` further down in the implementation, so probably better to align on modern pattern.